internalshared/configutil/listener_test.go

```diff
@@ -0,0 +1,49 @@
+package configutil
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestParseSingleIPTemplate(t *testing.T) {
+	type args struct {
+		ipTmpl string
+	}
+	tests := []struct {
+		name    string
+		args    args
```

Comment:
Since there's only one argument, is it worth defining this struct?  You could replace this field with `input string` and make it a little simpler.